### ISSUE TITLE
[IOS] Invalid Segment Alignment libmobileffmpeg.dylib does not have proper segment alignment. Try rebuilding the app with the latest Xcode version.

### ISSUE THREAD

**Issue by tanersener** [tanersener/mobile-ffmpeg#45]
share version build ok, but not upload

dylib error... my Xcode 10.1 version

<img width="748" alt="2018-11-18 11 00 28" src="https://user-images.githubusercontent.com/5547276/48667776-565a3280-eb21-11e8-85de-d59a5c215c8b.png">

_Originally posted by @thecsy in https://github.com/tanersener/mobile-ffmpeg/issues/18#issuecomment-439661736_

**Comment by tanersener** [comment]
@thecsy which `MobileFFmpeg` version do you use? Can you try using `v2.1.1`?

**Comment by tanersener** [comment]
@thecsy Can you confirm that Run/Test/Archive does not fail and this issue happens only on upload?

**Comment by thecsy** [comment]
Sorry, can't Xcode 9.4 

mobile-ffmpeg-full-3.0-ios-shared-universal
run :OK , Archive : capture error

mobile-ffmpeg-full-3.0-ios-static-universal
run fail , conflict libconv

mobile-ffmpeg-full-2.1.1-ios-shared-universal
run :OK , Archive : capture error

**Comment by thecsy** [comment]
test is Xcode 10.1 version use

yes , Invalid Segment Alignment' error

share version is always Error (Segment Alignment)

**Comment by tanersener** [comment]
Universal binaries are already tested with `test-app` and `Xcode 10.1`. `run` & `archive` both are successful. Are you sure you don't have another error?

<img width="980" alt="success" src="https://user-images.githubusercontent.com/10158439/48717167-e5eb0880-ec29-11e8-8dfb-2dff69f533be.png">

**Comment by thecsy** [comment]
please , Just one test. (appstore upload)
i can Arcive & build. however upload not work .

it's Build options to date
- 3.0  2.1, 2.2 pod   ( min version error )

<img width="701" alt="2018-11-29 10 22 09" src="https://user-images.githubusercontent.com/5547276/49225925-a0e48480-f428-11e8-911e-b8f74d3e2186.png">

- ios.sh build script universal version use ( alignment error )
- binary 3.0 share , 2.2 share ( alignment error )
- drop box link, version 3.0 share ( alignment error )

<img width="661" alt="2018-11-29 10 39 35" src="https://user-images.githubusercontent.com/5547276/49225915-9c1fd080-f428-11e8-9a67-a86ee674a2cb.png">


what's try option remain?

**Comment by tanersener** [comment]
Can you try with this [one](https://www.dropbox.com/s/u2bq66hf3obr8vx/mobile-ffmpeg-min-3.0-ios-shared-universal-2.zip)? This is different than the previous link.

**Comment by thecsy** [comment]
thanks, new link... 
but same error ( screen shot time see)


<img width="697" alt="2018-11-30 12 56 17" src="https://user-images.githubusercontent.com/5547276/49235703-02fbb480-f43e-11e8-82df-54326b326939.png">


you can upload? 
I deleted the suspicious pod contents.( VLC pod remove )
 But the results were the same.

**Comment by tanersener** [comment]
> thanks, new link...
> but same error ( screen shot time see)
> you can upload?
> I deleted the suspicious pod contents.( VLC pod remove )
> But the results were the same.

It was built with `Xcode 10.1`, which is the latest version. How is that possible?

Sorry but I don't have any other suggestions.

**Comment by codaman** [comment]
I copied this sentence from apple forum.

"The main problem is that you've built the framework to include binaries for both the simulator and devices. This isn't officially supported at all (because the iOS simulator is actually macOS platform, while devices are iOS platform). Your framework must contain only ARM architectures for iOS platform.
 
Try building the framework again with just the correct architectures and platform. It's possible all the other errors will go away too."

ffmpeg libraries is running also on intel architecture.

**Comment by codaman** [comment]
<img width="683" alt="ekran resmi 2018-12-17 02 39 37" src="https://user-images.githubusercontent.com/10552317/50060445-18464200-01a5-11e9-977e-168f23661044.png">
 may be we have to remove these intel  architectures.

**Comment by codaman** [comment]
yes finally I found the solution. Use static binary. there is a problems on framework and dynamic library. I successfully uploaded the store and tested on device.

**Comment by tanersener** [comment]
@codaman Yes, you really did it 👍 I was trying to find a solution for months and this `does not support the minimum OS Version specified in the Info.plist` message was so misleading. Thanks a lot my friend. You really saved my hours, days. I really appreciate that.

I'd like to ask if you can test the following solution for dynamic frameworks.

Can you add a new `Run Script Phase` after `Embed Frameworks` section and put the following lines inside? I'm expecting this to fix `dynamic frameworks`.

```
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libswresample.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libavfilter.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libavformat.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libavutil.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libswscale.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libavdevice.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/libavcodec.framework/strip-frameworks.sh"
bash "$BUILT_PRODUCTS_DIR/$FRAMEWORKS_FOLDER_PATH/mobileffmpeg.framework/strip-frameworks.sh"
```
<img width="922" alt="phases" src="https://user-images.githubusercontent.com/10158439/50068456-0c737380-01d7-11e9-89d7-aba991589944.png">

**Comment by codaman** [comment]
@tanersener i didn't try it. I will try it

**Comment by tanersener** [comment]
@codaman thanks my friend, I managed to test it but it didn't work, so don't waste your time if you haven't tried yet. I already posted the solution for #18.
